### PR TITLE
refactor vault provider instantiation

### PR DESCRIPTION
**Pull request by grantleehoffman**
* Change vault service client handler contract to simplify handler vars.
* Refactor Argo Cloud Ops service response codes.

### CODE REVIEW

**Review by pclata — DISMISSED** [review]
LGTM

**Review by thbishop-intuit — COMMENTED** [review]
just one question

File: service/handlers.go [hunk 1]
```diff
@@ -433,11 +417,17 @@ func (h handler) getWorkflowLogs(w http.ResponseWriter, r *http.Request) {
 
 	l := h.requestLogger(r, "op", "get-workflow-logs", "workflow", workflowName)
 
+	if err := validations.ValidateVar("workflow name", workflowName, "is_alphanumunderscore"); err != nil {
```
Comment (thbishop-intuit) [inline service/handlers.go:420]: does this come from argo workflows? what happens if they change the format (add or remove a char as being valid)?

Comment (grantleehoffman) [reply service/handlers.go:420]: Yeah it does, we had a test for this but we weren't validating it anywhere so I added this. Maybe remove this and the test then?

Comment (thbishop-intuit) [reply service/handlers.go:420]: yeah, i think dropping it makes sense. we're just a pass through for that data at this point

**Review by thbishop-intuit — APPROVED** [review]

**Pull request merged**